### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by itsmraga-hub**
# 💯  Achievements

- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should **not** be set by passing a parameter.
- Add authentication to your API endpoints.
- Your API endpoints should receive JSON and respond JSON as well.
- Login details
  - itsragamit@gmail.com
  - password: `qwertyuiop`
  - api_token: `69M6Zy6yPoxXKwrALpaG`

- Pass Api_token after url - ....?api_token=69M6Zy6yPoxXKwrALpaG

### CODE REVIEW

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @itsmraga-hub , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't show a contribution from your partner @stephnna  as this is a paired programming activity you should work on together.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by itsmraga-hub** [comment]
Hi @codecaiine , what do I do if my partner was a no-show?

**Comment by codecaiine** [comment]
Hi @itsmraga-hub Please reach out to the Student Success team. Also, note that you can go ahead alone with the project only if you receive approval from the Student Success team and post the screenshot of the approval mail from the Student Succes Team on the PR. Good luck :+1:

File: config/routes.rb [hunk 1]
```diff
@@ -11,6 +11,24 @@
...
 
+  resources :users, only: [] do
+    member do
+      get 'api_token'
+    end
+  end
+  # API routes
+  namespace :api do
+    namespace :v1 do
+      # post 'auth/login', to 'authenticate#login'
```
Comment (afanimartin) [inline config/routes.rb]: - Please remove all comments from the codebase. Comments make our code look ugly.

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @itsmraga-hub, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🎉 🎉 🎉 
- Requirements have been met

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (itsmraga-hub) [reply config/routes.rb]: Okay done.

**Review by OLIPLICHE — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# Highlights
- Applied the optional change requested by the review of the preview 👏 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by itsmraga-hub** [comment]
Thanks @OLIPLICHE. Will work on that @OLIPLICHE

**Pull request merged**